doc/CodeReview.md

```diff
@@ -77,7 +78,87 @@ possible responses:
 Core team members can change their votes at any time, based on discussion,
 changes, or further thought.
 
-## 3. Reviews
+## 3. Continuous Integration
+
+Tock leans heavily on automated integration testing and as a project is
+generally willing to explore new and novel means of testing for hardware
+reliability.
+
+With exceptions for drafts or works-in-progress, generally it is expected that
+a pull request pass the full continuous integration (CI) suite before core team
+members will perform significant review labor.
+
+One frequent challenge with CI setups is replicating failures in local
+development environments. Tock goes to great lengths to mitigate this as much
+as possible. Within reason, the inability to replicate a CI test in a local
+development environment shall be considered a bug (however, it is reasonable
+that local CI requires the install of non-trivial tooling, so long as there is
+a well-documented, reliable path to set up the tooling locally).
+
+### CI Organization
+
+All CI is driven by `make` rules.
+
+Generally, there are a series of fine-grained targets that do the actual tests,
+and then a meta layer of rules that are invoked depending on context.
+
+#### The short answer: `make ci`
+This is a meta-target that runs what Tock considers the "standard developer CI".
+This is the rule that should be run locally before submitting PRs.
+It runs the quicker jobs that catch the majority of small errors.
+
+
+#### The complete CI setup
+
+All CI is required to support the possibility of parallel make invocation (i.e.
+`make -jN`), but is not required to handle multiple independent make processes.
+
+##### `ci-job-*`
+To the extent reasonable, individual tests are broken into small atomic units.
+All actionable `make` recipes that run tests **must** be in `ci-job-*` rules.
+These perform individual, logical actions related only to testing.
+These rules **must not** perform any one-off or setup operations.
+No automated tooling should invoke job rules directly.
+If a CI check fails, developers should be able to run the failed `ci-job-*`
+locally, although in certain cases this may require installing supporting
+tooling.
+
+##### `ci-setup-*`
```

Comment:
Will `make ci` run `ci-setup-*` and then `ci-job-*`

Reply:
Effectively, thought the setup would be more of:
```make
ci-job-format-check: ci-setup-rustfmt
     run rustfmt

ci-setup-rustfmt:
     install rustfmt

ci:  ci-job-format-check
```